### PR TITLE
feat(rl): composable current-policy importance-sampling correction (TIS hook)

### PR DESCRIPTION
**Pull request by EazyReal**
## What

Makes off-policy importance sampling composable through the existing TIS correction hook by exposing the current policy log-probs to corrections.

- `policy_loss_function` now passes `cur_log_probs` (current πθ) into the correction kwargs, alongside frozen `train_log_probs` and `rollout_log_probs`. Existing `vanilla_tis_function` / `icepop_function` ignore it via `**kwargs`.
- `off_policy_is_function`: truncated IS with detached weight `clip(πθ / π_rollout)` using `--eps-clip` / `--eps-clip-high`. One weight corrects train/inference mismatch and async staleness.

Independent of #1343 (based on `origin/main`). On a REINFORCE base this reproduces CISPO:

```
--advantage-estimator reinforce --use-tis \
  --custom-tis-function-path miles.backends.training_utils.loss_hub.corrections.off_policy_is_function
```

Canonical single-sided CISPO: `--eps-clip 1.0`. No named CISPO estimator.

## Tests

`tests/fast/backends/training_utils/loss/test_off_policy_is.py`.

### CODE REVIEW

File: miles/backends/training_utils/loss_hub/corrections.py [hunk 1]
```diff
@@ -60,3 +60,33 @@ def icepop_function(
...
+    """Off-policy truncated IS: like `vanilla_tis_function` but with the *current*
+    policy in the numerator instead of the old recompute, so the (detached) weight is
+    `clip(pi_theta / pi_rollout)` against the actual rollout logprob -- one weight that
+    corrects both the train/inference mismatch and async (multi-version) staleness.
+    Composed with `--advantage-estimator reinforce` it is the CISPO surrogate
+    (https://arxiv.org/abs/2506.13585); `--eps-clip 1.0` gives canonical single-sided
+    clipping. Same `(pg_loss, loss_masks, metrics)` contract; `loss_masks` unchanged.
+    """
+    cur = torch.cat([lp.detach() for lp in cur_log_probs], dim=0)
+    rollout = torch.cat(rollout_log_probs, dim=0)
```
[the remark this answers is not in the document]
Comment (EazyReal) [reply miles/backends/training_utils/loss_hub/corrections.py:83]: Thanks. Leaving this as-is for consistency: `off_policy_is_function` mirrors the existing `vanilla_tis_function` / `icepop_function` in this module, which `torch.cat` the same per-sample lists without length/shape asserts. `cur_log_probs` and `rollout_log_probs` are built together from the same batch by the caller, and a genuine mismatch would surface immediately at `cur - rollout`. Happy to add up-front validation across all the correction hooks as a separate consistency pass if you'd prefer that.

**Comment by EazyReal** [comment]
@fzyzcjy @yueming-yuan @maocheng23 @Zhichenzzz @Shi-Dong Could I get a quick review on this current-policy IS hook? It makes rollout-vs-current correction composable instead of hardwiring a separate estimator path.
